contract_digitized_signature/models/account_analytic_account.py

```diff
@@ -12,6 +12,11 @@ class AccountAnalyticAccount(models.Model):
     customer_signature = fields.Binary(
         string='Customer acceptance',
     )
+    signature_name = fields.Many2one(
```

Comment:
IMO this should be a plain text field, possibly containing a capture of the name from the partner (not sure how your workflow works). Linking a signature to a dynamic record means that the actual data can change out from underneath the legal data

Reply:
Well, you shouldn't modify contacts when there are changes, but deactivate them and create new ones. The problem with the plain text is that requires to fully type it. With this, you can select directly by direct search.

Reply:
@pedrobaeza - none of my customers do that, so while yeah it's the best option, it's still not a legal field.

An alternative would be to add a `signatory_id` column that is the relation, with an  onchange to update the `signature_name`.

Furthermore- this creates an incompatibility with the way Odoo does it, which is the way we did it. The goal of adding this was to make our addons live in harmony. Adding this, but making it a relation, takes us further from that goal than simply not having the column

Reply:
Also - to be a relation, it needs to be prefixed with `_id`, which is not the case with `signature_name`

Reply:
OK, let's change it to a char field. At the end, we don't have that requirement.

Reply:
done!